smart-send-logistics/includes/frontend/class-ss-shipping-frontend.php

```diff
@@ -121,25 +189,34 @@ public function display_ss_pickup_points($method, $index)
          * @param $street string
          *
          * @return array
-	     */
-        public function find_closest_agents_by_address($carrier, $country, $postal_code, $city, $street)
+         */
+        public function find_closest_agents_by_address($carrier, $country, $postal_code, $city, $street, $restapiend)
```

Comment:
What is this extra parameter that you just added to the function?

Reply:
The additional parameter restapiend was introduced to the find_closest_agents_by_address function to address a specific challenge related to session management in WordPress.
In the context of WordPress endpoints, particularly when working with REST API requests, the usual WooCommerce (WC) sessions are not available or may not be reliably used. This limitation necessitates an alternative approach for handling session data. By incorporating the restapiend parameter, we can differentiate between requests that are made through WordPress REST API endpoints and those that are not.
The presence of this parameter allows the function to determine whether the request originates from an API endpoint. Based on this, the function can then decide whether to utilise Woo-commerce sessions or fallback to PHP sessions as appropriate. This approach ensures that session data is managed correctly and consistently, regardless of the request context.

Reply:
updated.